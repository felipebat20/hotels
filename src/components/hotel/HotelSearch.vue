<template>
  <q-card class="bg-white rounded-borders-md">
    <q-card-section class="q-px-md q-py-sm">
      <h2 class="text-h5 q-ma-none">Reservar hotel</h2>
    </q-card-section>

    <q-separator />

    <q-card-section class="column q-gutter-md">
      <div>
        <label>
          <span>
            Destino
          </span>

          <q-select
            v-model="city_search"
            :options="filteredPlaces"
            use-input
            hide-selected
            fill-input
            dense
            outlined
            hide-dropdown-icon
            input-debounce="0"
            placeholder="Cidade, UF"
            option-value="placeId"
            class="place-select"
            @filter="handleFilterPlaces"
          />
        </label>
      </div>

      <div class="row justify-end">
        <q-btn
          rounded
          no-caps
          color="primary"
          size="md"
          class="col-12 col-sm-auto"
          style="min-width: 180px"
        >
          Buscar

          <template v-slot:loading>
            <q-spinner-hourglass class="on-left" />

            Buscando...
          </template>
        </q-btn>
      </div>
    </q-card-section>
  </q-card>
</template>

<script setup lang="ts">
import { ref, computed } from 'vue';

import places from '@/data/place.json';

const city_search = ref('');

const options = ref(places);

const filteredPlaces = ref();

const getFormattedPlaces = computed(() => {
  return options.value.map((city) => {
    return {
      label: `${city.name}, ${city.state.name}`,
      value: city.placeId,
      ...city
    };
  });
});


const handleFilterPlaces = (value: string, update: (callbackFn: () => void) => void) => {
  update(() => {
    const new_search = value.toLowerCase();

    filteredPlaces.value = getFormattedPlaces.value.filter((place) => {
      return place.label.toLowerCase().includes(new_search) || place.state.name.toLowerCase().includes(new_search);
    })
  });
};
</script>
