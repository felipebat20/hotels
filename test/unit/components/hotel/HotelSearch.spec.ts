import { installQuasarPlugin } from '@quasar/quasar-app-extension-testing-unit-vitest';
import { mount } from '@vue/test-utils';
import { describe, expect, it, vi } from 'vitest';
import { createTestingPinia } from '@pinia/testing'

import HotelSearch from '@/components/hotel/HotelSearch.vue';

installQuasarPlugin();

describe('test HotelSearch', () => {
  const createWrapper = () => {
    return mount(HotelSearch, {
      global: {
        plugins: [createTestingPinia({ createSpy: vi.fn })],
      },
    });
  };

  it('should mount component properly', () => {
    const wrapper = createWrapper();
    const html = wrapper
      .html()
      .replace(/id="f_[^"]+"/g, 'id="static-id"')
      .replace(/for="f_[^"]+"/g, 'for="static-for"')
      .replace(/aria-controls="f_[^"]+"/g, 'aria-controls="static-aria-controls"')

    expect(html).toMatchSnapshot();
  });
});
