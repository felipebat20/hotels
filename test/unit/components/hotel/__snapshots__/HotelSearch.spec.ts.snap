// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`test HotelSearch > should mount component properly 1`] = `
"<div data-v-5876535b="" class="q-card bg-white no-shadow border-radius-md">
  <div data-v-5876535b="" class="q-card__section q-card__section--vert q-px-md q-py-sm">
    <h2 data-v-5876535b="" class="text-h5 q-ma-none">Reservar hotel</h2>
  </div>
  <hr data-v-5876535b="" class="q-separator q-separator--horizontal" aria-orientation="horizontal">
  <div data-v-5876535b="" class="q-card__section q-card__section--vert column q-gutter-md">
    <div data-v-5876535b=""><label data-v-5876535b=""><span data-v-5876535b=""> Destino </span><label data-v-5876535b="" class="q-field row no-wrap items-start q-field--outlined q-select q-field--auto-height q-select--with-input q-select--without-chips q-select--single q-field--dense place-select" for="static-for">
          <!---->
          <div class="q-field__inner relative-position col self-stretch">
            <div class="q-field__control relative-position row no-wrap" tabindex="-1">
              <div class="q-field__control-container col relative-position row no-wrap q-anchor--skip">
                <div class="q-field__native row items-center"><input class="q-field__input q-placeholder col" type="search" tabindex="0" role="combobox" aria-readonly="false" aria-autocomplete="list" aria-expanded="false" aria-controls="static-aria-controls" placeholder="Cidade, UF" id="static-id" value=""></div>
                <!---->
              </div>
              <!---->
              <!---->
            </div>
            <!---->
          </div>
          <!---->
        </label></label></div>
    <div data-v-5876535b=""><label data-v-5876535b=""><span data-v-5876535b=""> Destino </span><label data-v-5876535b="" class="q-field row no-wrap items-start q-field--outlined q-input q-field--dense full-width" for="static-for">
          <!---->
          <div class="q-field__inner relative-position col self-stretch">
            <div class="q-field__control relative-position row no-wrap" tabindex="-1">
              <div class="q-field__control-container col relative-position row no-wrap q-anchor--skip"><input class="q-field__native q-placeholder" tabindex="0" placeholder="Nome do Hotel" required="" id="static-id" type="text" value="">
                <!---->
              </div>
            </div>
            <!---->
          </div>
          <!---->
        </label></label></div>
    <div data-v-5876535b="" class="row justify-end"><button data-v-5876535b="" class="q-btn q-btn-item non-selectable no-outline q-btn--standard q-btn--rectangle q-btn--rounded bg-primary text-white q-btn--actionable q-focusable q-hoverable q-btn--no-uppercase col-12 col-sm-auto" style="font-size: 14px; min-width: 180px;" tabindex="0" type="button"><span class="q-focus-helper"></span><span class="q-btn__content text-center col items-center q-anchor--skip justify-center row">Buscar </span>
        <transition-stub name="q-transition--fade" appear="false" persisted="false" css="true">
          <!---->
        </transition-stub>
      </button></div>
  </div>
</div>"
`;
